pkg/pkgmgr/rpm.go

```diff
@@ -599,99 +600,142 @@ func (rm *rpmManager) unpackAndMergeUpdates(ctx context.Context, updates unversi
 	//  - Reports being slightly out of date, where a newer security revision has displaced the one specified leading to not found errors.
 	//  - Reports not specifying version epochs correct (e.g. bsdutils=2.36.1-8+deb11u1 instead of with epoch as 1:2.36.1-8+dev11u1)
 	//  - Reports specifying remediation packages for cbl-mariner v1 instead of v2 (e.g. *.cm1.aarch64 instead of *.cm2.aarch64)
 	var downloadCmd string
+	const queryFormat = `--queryformat "%s" %s > "%s"`
+
+	pkgs := ""
+
 	if updates != nil {
-		const rpmDownloadTemplate = `yumdownloader --downloadonly --downloaddir=. --best -y %s`
+		rpmDownloadTemplate := `
+		set -x
+		packages="%s"
+		echo "$packages"
+
+		mkdir -p /tmp/rootfs/var/lib
+		ln -s /tmp/rpmdb /tmp/rootfs/var/lib/rpm
+
+		rpm --dbpath=/tmp/rootfs/var/lib/rpm -qa
+
+		for package in $packages; do
+			package="${package%%.*}" # trim anything after the first "."
+			output=$(tdnf install -y --releasever=$OS_VERSION --installroot=/tmp/rootfs ${package} 2>&1)
+
+			if [ "$IGNORE_ERRORS" = "false" ] && [ $? -ne 0 ]; then
+				exit $?
+			fi
+		done
+
+		mkdir /tmp/rootfs/var/lib/rpmmanifest
+
+		rpm --dbpath=/tmp/rootfs/var/lib/rpm -qa | tee /tmp/rootfs/var/lib/rpmmanifest/container-manifest-1
+	
+		echo "testing just name"
+		rpm --dbpath=/tmp/rootfs/var/lib/rpm -qa --qf \"\%{NAME}\"
+
+		rpm --dbpath=/tmp/rootfs/var/lib/rpm -qa
+		rm /tmp/rootfs/var/lib/rpm
+		rm -rf /tmp/rootfs/var/cache/tdnf
+	
+		rpm --dbpath /tmp/rpmdb -qa ` + fmt.Sprintf(queryFormat, resultQueryFormat, pkgs, "/tmp/rootfs/manifest") + ``
+
 		pkgStrings := []string{}
 		for _, u := range updates {
 			pkgStrings = append(pkgStrings, u.Name)
 		}
+
 		downloadCmd = fmt.Sprintf(rpmDownloadTemplate, strings.Join(pkgStrings, " "))
 	} else {
-		// only updated the outdated pacakges from packages.txt
-		downloadCmd = `
+		// only updated the outdated packages from packages.txt
+		downloadCmd = `		
+		set -x
+
 		packages=$(<packages.txt)
+		echo "$packages"
+		mkdir -p /tmp/rootfs/var/lib
+		ln -s /tmp/rpmdb /tmp/rootfs/var/lib/rpm
+
+		rpm --dbpath=/tmp/rootfs/var/lib/rpm -qa
 		for package in $packages; do
-			output=$(yumdownloader --downloadonly --downloaddir=. --best -y "$package" 2>&1)
-			if [ $? -ne 0 ]; then
-				echo "$output" >>error_log.txt
+			package="${package%%.*}" # trim anything after the first "."
+			output=$(tdnf install -y --releasever=$OS_VERSION --installroot=/tmp/rootfs ${package} 2>&1)
+
+			if [ "$IGNORE_ERRORS" = "false" ] && [ $? -ne 0 ]; then
+				exit $?
 			fi
 		done
+
+		mkdir /tmp/rootfs/var/lib/rpmmanifest
+		
+		rpm --dbpath=/tmp/rootfs/var/lib/rpm -qa | tee /tmp/rootfs/var/lib/rpmmanifest/container-manifest-1
+		rpm --dbpath=/tmp/rootfs/var/lib/rpm -qa --qf '%{NAME}\t%{VERSION}-%{RELEASE}\t%{INSTALLTIME}\t%{BUILDTIME}\t%{VENDOR}\t%{EPOCH}\t%{SIZE}\t%{ARCH}\t%{EPOCHNUM}\t%{SOURCERPM}\n' | grep -v '^gpg-pubkey' | tee /tmp/rootfs/var/lib/rpmmanifest/container-manifest-2
```

Comment:
we can do `rpm --dbpath=/tmp/rootfs/var/lib/rpm --erase --allmatches gpg-pubkey-*` instead of ignoring